src/text-editor.coffee

```diff
@@ -1561,6 +1561,16 @@ class TextEditor extends Model
   setCursorBufferPosition: (position, options) ->
     @moveCursors (cursor) -> cursor.setBufferPosition(position, options)
 
+  # Essential: Get a {Cursor} at given screen coordinates {Point}
+  #
+  # * `position` A {Point} or {Array} of `[row, column]`
+  #
+  # Returns the first matched {Cursor} or undefined
+  getCursorAtScreenPosition: (position) ->
+    for cursor in @cursors
+      return cursor if cursor.getScreenPosition().isEqual(position)
+    undefined
```

Comment:
I would say don't use `return` here, just write either `undefined` or `null`. The one gotcha with coffee-script is that you can't just remove the last line: if the last statement in a function is a `for` loop, `coffee-script` will use that loop as a *value*, collecting its results into an array.

`null` and `undefined` are both fine. Sometimes I use `undefined` for this purpose, to be consistent with `object["non-existent-key"]` or `array[outOfBoundsIndex]`.